packages/medusa/src/subscribers/payment-webhook.ts

```diff
@@ -0,0 +1,48 @@
+import { PaymentWebhookEvents } from "@medusajs/utils"
+
+import {
+  IEventBusService,
+  IPaymentModuleService,
+  ProviderWebhookPayload,
+  Subscriber,
+} from "@medusajs/types"
+import { EventBusService } from "../services"
+
+type SerializedBuffer = {
+  data: ArrayBuffer
+  type: "Buffer"
+}
+
+type InjectedDependencies = {
+  paymentModuleService: IPaymentModuleService
+  eventBusService: EventBusService
+}
+
+class PaymentWebhookSubscriber {
+  private readonly eventBusService_: IEventBusService
+  private readonly paymentModuleService_: IPaymentModuleService
+
+  constructor({ eventBusService, paymentModuleService }: InjectedDependencies) {
+    this.eventBusService_ = eventBusService
+    this.paymentModuleService_ = paymentModuleService
+
+    this.eventBusService_.subscribe(
+      PaymentWebhookEvents.WebhookReceived,
+      this.processEvent as Subscriber
+    )
+  }
+
+  /**
+   * TODO: consider moving this to a workflow
+   */
+  processEvent = async (data: ProviderWebhookPayload): Promise<void> => {
+    if ((data.payload.data as unknown as SerializedBuffer).type === "Buffer") {
+      data.payload.data = Buffer.from(
+        (data.payload.data as unknown as SerializedBuffer).data
+      )
+    }
```

Comment:
@fPolic, why was this added here?

Reply:
Since we are dispatching the webhook payload via the event bus, the body, which is a buffer, is serialized. Stripe expects a buffer or a string of the request body so we are reconstructing a buffer in the subscriber.

Alternatively, one way of unifying this across provides could be to always pass the raw body(string/buffer) to a plugin which would be responsible for parsing the webhook payload?

Reply:
I see, makes sense.

> Alternatively, one way of unifying this across provides could be to always pass the raw body(string/buffer) to a plugin

This sounds like a reasonable approach to me. I would perhaps even add both the raw body and parsed body, so providers generally wouldn't need to do anything with the payload. I believe the way Stripe handles it, it actually an edge case, which is weird coming from them but I am sure they have good reasons.

So in practice, I would remove the `bodyParser` from the middleware and add a custom middleware that attaches both types of parsed payloads. The API route would end up looking something like:
```
export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
  try {
   ...
    const event = {
      provider,
      payload: { data: req.body, headers: req.headers, rawData: req.rawBody },
    }
  ...
```

Wdyt?

Reply:
@olivermrbl - I've added support for `rawBody` but with a slightly different approach in [this](https://github.com/medusajs/medusa/pull/6311/commits/4ec6d16b963bd1e276d542d196ba2a9eb2c24e93) commit. Let me know wdyt.

Reply:
This looks good to me. @kasperkristensen, can you glance over this commit too? 🙏

Reply:
You mean the deserialization that we do in the subscriber? Since we have a serialized buffer because we transmit data through the event bus I think we should "fix" the data back in the subscriber. From the plugin perspective, I'm not sure why we would expect to receive a serialized buffer instead of a plain buffer there.